### PR TITLE
Convert to a proper package

### PR DESCRIPTION
**Pull request by remyperona**
- Rename namespace to `WPMedia\Options`
- Add `OptionsInterface` as the base interface of the package
- Update `AbstractOptions` to implement `OptionsInterface`
- Add `SiteOptions` to manage multisite options, extending `AbstractOptions`
- Rename & Update `OptionArray` to remove hardcoded references to WP Rocket
- Update all existing tests
- Add new tests
- Cleanup config files
- Update readme with more details

### CODE REVIEW

File: OptionArray.php [hunk 1]
```diff
@@ -1,27 +1,36 @@
...
 
 /**
- * Manages the data inside an option.
+ * Manages the array data coming from an option.
  *
  * @since 3.0
  * @author Remy Perona
  */
-class Options_Data {
+class OptionArray {
```
Comment (hellofromtonya) [inline OptionArray.php]: @Tabrisrp Check out implementing `ArrayAccess`, which allows the objects to access it as an array. It then fits with the new name `OptionArray`.

```php
use AccessArray;

class OptionArray implements AccessArray {
```

[See the docs here](https://www.php.net/manual/en/class.arrayaccess.php).

You'll need to add 4 methods to allow array iteration:

- `offsetSet`
- `offsetExists`
- `offsetUnset`
- `offsetGet`

Laravel is liberal in its use of `ArrayAccess`. Check out how they do it with things like the [`Container`](https://github.com/laravel/framework/blob/0a70bebd5ecfd51185a312bbfb60ee7f8ff7eb09/src/Illuminate/Container/Container.php#L15), [`Config`](https://github.com/laravel/framework/blob/0a70bebd5ecfd51185a312bbfb60ee7f8ff7eb09/src/Illuminate/Config/Repository.php#L9) and [`Cache\Repository`](https://github.com/laravel/framework/blob/0a70bebd5ecfd51185a312bbfb60ee7f8ff7eb09/src/Illuminate/Cache/Repository.php).

Comment (remyperona) [reply OptionArray.php]: Would you implement it in addition to the methods already existing in the class, or in replacement of them?

Comment (hellofromtonya) [reply OptionArray.php]: In addition to.... These new ones call the existing methods.

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,2 @@
-/node_modules/
 /vendor/*
-/tests/env/local/
+composer.lock
```
Comment (iCaspar) [inline .gitignore:2]: I think we said we were going to track the lock files in the repos now. Is this change left from a while ago before we changed that practice?

**Review by iCaspar — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 2 above)
Comment (remyperona) [reply .gitignore:2]: On that, I'm basing this on composer documentation, notably:
> For libraries it is not necessary to commit the lock file. this lock file will not have any effect on other projects that depend on it. It only has an effect on the main project.

It was also causing CI issues when committed, so I opted to ignore it

Comment (iCaspar) [reply .gitignore:2]: Alright. 👍

**Review by iCaspar — APPROVED** [review]

**Pull request merged**